core/src/main/java/org/apache/accumulo/core/client/admin/NewTableConfiguration.java

```diff
@@ -146,4 +161,75 @@ public NewTableConfiguration enableSummarization(SummarizerConfiguration... conf
     summarizerProps = tmp;
     return this;
   }
+
+  /**
+   * Configures a table's locality groups prior to initial table creation.
+   *
+   * Allows locality groups to be set prior to table creation. Additional calls to this method prior to table creation will overwrite previous locality group
+   * mappings.
+   *
+   * @param groups
+   *          mapping of locality group names to column families in the locality group
+   *
+   * @since 2.0.0
+   *
+   * @see TableOperations#setLocalityGroups
+   */
+  public NewTableConfiguration setLocalityGroups(Map<String,Set<Text>> groups) {
+    // ensure locality groups do not overlap
+    LocalityGroupUtil.ensureNonOverlappingGroups(groups);
+    localityProps = new HashMap<>();
+    for (Entry<String,Set<Text>> entry : groups.entrySet()) {
+      Set<Text> colFams = entry.getValue();
+      String value = LocalityGroupUtil.encodeColumnFamilies(colFams);
+      localityProps.put(Property.TABLE_LOCALITY_GROUP_PREFIX + entry.getKey(), value);
+    }
+    localityProps.put(Property.TABLE_LOCALITY_GROUPS.getKey(), groups.keySet().stream().collect(Collectors.joining(",")));
+    return this;
+  }
+
+  /**
+   * Configure iterator settings for a table prior to its creation.
+   *
+   * Additional calls to this method before table creation will overwrite previous iterator settings.
+   *
+   * @param setting
+   *          object specifying the properties of the iterator
+   * @throws AccumuloException
+   *           if a general error occurs
+   *
+   * @since 2.0.0
+   */
+  public NewTableConfiguration attachIterator(IteratorSetting setting) throws AccumuloException {
+    return attachIterator(setting, EnumSet.allOf(IteratorScope.class));
+  }
+
+  /**
+   * Configure iterator settings for a table prior to its creation.
+   *
+   * @param setting
+   *          object specifying the properties of the iterator
+   * @param scopes
+   *          enumerated set of iterator scopes
+   * @throws AccumuloException
+   *           if a general error occurs
+   *
+   * @since 2.0.0
+   */
+  public NewTableConfiguration attachIterator(IteratorSetting setting, EnumSet<IteratorScope> scopes) throws AccumuloException {
+    checkArgument(setting != null, "setting is null");
+    checkArgument(scopes != null, "scopes is null");
```

Comment:
Would `Objects.requiresNonNull()` be better here?

Reply:
Yes. Like that better. Perhaps should update the TableOperationsHelper attachIterator version as well.

Reply:
Probably the reason it wasn't changed in TableOperationsHelper is because it's API and a change in behavior from `IllegalArgumentException` to `NullPointerException`. I wouldn't change the existing API. It's up to you whether the new API should be "clean" (NPE), or model the former (IAE).